src/main/java/com/internship/backend/service/UserService.java

```diff
@@ -4,119 +4,123 @@
-
         authorityRepository.save(iterator.next());
 
         return savedUser;
     }
 
     public Users fromDTO(UserDTO userDTO) {
-        Users user = Users.builder()
-                .username(userDTO.getUsername())
-                .password(userDTO.getPassword())
-                .email(userDTO.getEmail())
-                .build();
-
-        Set<Authority> authorities = new HashSet<>();
-
-        Authority authority = new Authority();
-        authority.setName(userDTO.getRole());
-        authority.setUser(user);
+        Users user = new Users();
+        user.setUsername(userDTO.getUsername());
+        user.setPassword(userDTO.getPassword());
+        user.setEmail(userDTO.getEmail());
 
-        authorities.add(authority);
-        user.setAuthorities(authorities);
+        Authority authority = resolveAuthority(userDTO.getRole());
+        user.setAuthorities(Collections.singleton(authority));
 
         return user;
     }
 
-    public Users update(int userId, Users updatedUser) throws UserDoesNotExistException {
-        Users user = userRepository.findById(userId).orElseThrow(() -> new UserDoesNotExistException("User not found"));
+    private Authority resolveAuthority(String roleName) {
+        Optional<Authority> optionalAuthority = authorityRepository.findByName(roleName);
+        if (optionalAuthority.isPresent()) {
+            return optionalAuthority.get();
+        } else {
+            Authority newAuthority = new Authority(roleName);
+            newAuthority.setId(idGeneratorService.getCurrentId());
+            return authorityRepository.save(newAuthority);
+        }
+    }
 
-        user.setUsername(updatedUser.getUsername());
-        user.setPassword(updatedUser.getPassword());
-        user.setEmail(updatedUser.getEmail());
 
-        return userRepository.save(user);
+    public List<Users> getAllUsers() {
+        List<Users> users = userRepository.findAll();
+
+        users.forEach(user -> {
+            List<Reservation> reservations = reservationRepository.findByUserId(user.getId());
+            user.setReservations(reservations);
+        });
+        logger.info("Number of users: " + users.size());
+        return users;
     }
 
-    public void delete(int userId) throws UserDoesNotExistException {
-        if (!userRepository.existsById(userId))
-            throw new UserDoesNotExistException("User does not exist");
 
-        userRepository.deleteById(userId);
-
-        if (userRepository.count() == 0) {
-            userRepository.resetAutoIncrementId();
-        }
+    public Optional<Users> getUserById(Integer id) {
+        return userRepository.findById(id);
     }
 
-    public Users login(String username, String password) {
-        var users = userRepository.findByUsername(username);
 
-        if (!ObjectUtils.isEmpty(users))
-            return users;
-
-        return null;
+    public Users updateUser(Integer id, Users userDetails) {
```

Comment:
this method could be rewritten : 
 return userRepository.findById(id)
                .map(user -> {
                    user.setUsername(userDetails.getUsername());
                    user.setPassword(userDetails.getPassword());
                    user.setEmail(userDetails.getEmail());
                    user.setAuthorities(userDetails.getAuthorities());
                    return userRepository.save(user);
                })
                .orElseThrow(() -> new RuntimeException("User not found with id " + id));
Maybe you can try to extract what's inside the map in a separate method